src/actions/VaccineActions.js

```diff
@@ -38,13 +40,39 @@ const startSensorScan = () => async (dispatch, getState) => {
     return null;
   }
 
-  dispatch(scanForSensors());
+  await BleService()
+    .blinkWithRetries(macAddress, 3)
+    .catch(error => {
+      const { message } = error;
+      ToastAndroid.show(message || error, ToastAndroid.LONG);
+    });
 
-  // Scan will continue running until it is stopped...
   return null;
 };
 
-const scanForSensors = () => async (dispatch, getState) => {
+const startSensorScan = () => async (dispatch, getState) => {
+  const bluetoothEnabled = PermissionSelectors.bluetooth(getState());
+  const locationPermission = PermissionSelectors.location(getState());
+
+  // Ensure the correct permissions before initiating a new sync process.
+  if (!bluetoothEnabled) await dispatch(PermissionActions.requestBluetooth());
+  if (!locationPermission) await dispatch(PermissionActions.requestLocation());
+
+  if (!bluetoothEnabled) {
+    ToastAndroid.show(syncStrings.bluetooth_disabled, ToastAndroid.LONG);
+    return null;
+  }
+
+  if (!locationPermission) {
```

Comment:
I wonder if we can encapsulate this within its own thunk.. probably used in the upcoming thunks also?

Reply:
absolutely! great idea